db/changes.go

```diff
@@ -587,7 +592,11 @@ func (db *Database) checkForUserUpdates(ctx context.Context, userChangeCount uin
 
 			changedRoles := db.user.RoleNames().CompareKeys(previousRoles)
 			if len(changedRoles) > 0 {
-				changeWaiter.RefreshUserKeys(db.User())
+				collectionID, err := db.getCollectionID()
+				if err != nil {
+					return false, 0, nil, err
+				}
+				changeWaiter.RefreshUserKeys(db.User(), collectionID)
```

Comment:
This reinforces the idea that changeWaiter should just be working with strings - there isn't any need for a collection when talking about user/role keys in the changeWaiter - these are still just db-scoped (not collection-scoped) users and roles that need to be notified when their user/role documents change.